src/lib/comms/include/sol-http-server.h

```diff
@@ -204,10 +214,99 @@ int sol_http_server_set_last_modified(struct sol_http_server *server, const char
  * and parameters (e.g http headers)
  *
  * @return @c 0 on success, error code (always negative) otherwise.
+ *
+ * @see sol_http_server_send_response_progressive
  */
 int sol_http_server_send_response(struct sol_http_request *request, struct sol_http_response *response);
 
 /**
+ * @brief Set the necessary headers to allow server sent events
+ *
+ * @param response The response to set the headers
+ *
+ * @return @c 0 on success, error code (always negative) otherwise.
+ */
+static inline int
+sol_http_response_set_progressive_headers(struct sol_http_response *response)
+{
+    if (!response)
+        return -EINVAL;
+
+    if (!sol_http_param_add(&response->param,
+        SOL_HTTP_REQUEST_PARAM_HEADER("Content-Type", "text/event-stream"))) {
+        return -1;
```

Comment:
also, I'm thinking since these are all the same pattern if you should do a loop:

```c
const struct { const char *k, const char *v } *itr, headers[] = { // or make it of struct sol_http_param
   { "Content-type", "text/event-stream" },
   { "Connection", "keep-alive" },
   { "Cache-Control", "no-cache" },
   { NULL, NULL }
};
for (itr = headers; itr->k != NULL; itr++) {
    if (!sol_http_param_add(&response->param, SOL_HTTP_REQUEST_PARAM_HEADER(itr->k, itr->v))) { ... }
}
```

You can even make this a helper function in sol-http.h (or server.h) and use it everywhere, then you take a null-terminated array and loop-set it. Or make a va_args version that is null terminated. This pattern is very common and I've seen it in many places.